uwblueprint/paramedics-react
# Patient Restore Flow

[Notion ticket](https://www.notion.so/uwblueprintexecs/As-a-user-I-want-the-flexibility-to-delete-a-patient-entry-f61270edfb854f21a4d1302a247933de)

[Notion ticket 2](https://www.notion.so/uwblueprintexecs/Remove-Edit-Button-for-Restore-Flow-from-profile-pop-up-72bc6ad548504c5088a158418b9a5641)

[Figma link](https://www.figma.com/file/IpOPyv79gBwEohvOJvPV6f/%F0%9F%9A%91-Designs-for-Development-S20-F20?node-id=1953%3A0)


## Changes
- Created restore flow (e.g. modals, greyed out rows, and restore menu) for patients
- Changed title, delete button, and default status field for patient profile page
- Added loading state to patient profile page
- Added snack bar for patient restore notification
- Replaced edit with restore when clicking on a deleted patients row (restore modal pops up when restore is clicked)


## Screenshots

<img width="1136" alt="Screen Shot 2020-11-19 at 5 36 09 PM" src="https://user-images.githubusercontent.com/59406513/99747632-cc3a6100-2a8f-11eb-946f-54592c67d7cd.png">
estore modal pops up when clicked)

<img width="1429" alt="Screen Shot 2020-11-03 at 7 22 58 PM" src="https://user-images.githubusercontent.com/59406513/98065735-fd692f00-1e09-11eb-8f72-70dc720e20b5.png">

<img width="1223" alt="Screen Shot 2020-11-03 at 7 23 18 PM" src="https://user-images.githubusercontent.com/59406513/98065753-08bc5a80-1e0a-11eb-99df-7c80ba5e4388.png">
<img width="1267" alt="Screen Shot 2020-11-03 at 7 23 31 PM" src="https://user-images.githubusercontent.com/59406513/98065767-11149580-1e0a-11eb-8a9e-3e62d8dcedb1.png">
<img width="1440" alt="Screen Shot 2020-11-03 at 7 23 51 PM" src="https://user-images.githubusercontent.com/59406513/98065799-1c67c100-1e0a-11eb-85f2-1287f970d73a.png">
<img width="1440" alt="Screen Shot 2020-11-03 at 7 24 05 PM" src="https://user-images.githubusercontent.com/59406513/98065809-24276580-1e0a-11eb-9fae-7cdaf4370273.png">
<img width="1440" alt="Screen Shot 2020-11-03 at 7 24 12 PM" src="https://user-images.githubusercontent.com/59406513/98065819-28538300-1e0a-11eb-931c-d87d86f6d0ff.png">
<img width="1439" alt="Screen Shot 2020-11-03 at 7 25 05 PM" src="https://user-images.githubusercontent.com/59406513/98065892-5df86c00-1e0a-11eb-9de0-cfa6212e872b.png">

## Testing

Test the following things and see if it matches the screenshots:
- Open up to the CCP Dashboard Page
- Delete patients to see if the corresponding rows are greyed out (add patients if needed)
- See if each deleted patient is separated from the active patients
- Try to open up the option menus and see if the restore modal pops up
- See if you cancel the modal
- See if you can confirm the restoration of the patient
- Check out if the patient profile pages matches the changes and additions
- Make changes and click complete to see if the changes are reflected for each status
- See if the snack bar with the correct info pops up


## Checklist
>[Code review doc for reference](https://www.notion.so/uwblueprintexecs/Code-Review-a21fc85b00394f488e92d9d605f6b2bc)

before opening PR

filling out PR

after opening PR
